tests/js/sentry-test/performance/utils.ts

```diff
@@ -0,0 +1,183 @@
+    const {span} = mSpan;
+    span.span_id = spanId;
+    span.trace_id = this.TRACE_ID;
+    span.parent_span_id = parentSpanId ?? this.ROOT_SPAN_ID;
+
+    this.#event.entries[0].data.push(span);
+
+    switch (mSpan.problemSpan) {
+      case ProblemSpan.PARENT:
+        this.#event.perfProblem?.parentSpanIds.push(spanId);
+        break;
+      case ProblemSpan.OFFENDER:
+        this.#event.perfProblem?.offenderSpanIds.push(spanId);
+        break;
+      default:
+        break;
+    }
+
+    if (span.timestamp > this.#event.endTimestamp) {
+      this.#event.endTimestamp = span.timestamp;
+    }
+
+    mSpan.children.forEach(child => this.addSpan(child, spanId));
+  }
+
+  getEvent() {
+    return this.#event;
+  }
+}
+
+/**
+ * A MockSpan object to be used for testing. This object is intended to be used in tandem with `TransactionEventBuilder`
+ */
+export class MockSpan {
+  span: RawSpanType;
+  children: MockSpan[] = [];
+  problemSpan: ProblemSpan | undefined;
+
+  /**
+   *
+   * @param opts.startTimestamp
+   * @param opts.endTimestamp
+   * @param opts.op The operation of the span
+   * @param opts.description The description of the span
+   * @param opts.status Optional span specific status, defaults to 'ok'
+   * @param opts.problemSpan If this span should be part of a performance problem, indicates the type of problem span (i.e ProblemSpan.OFFENDER, ProblemSpan.PARENT)
+   * @param opts.parentSpanId When provided, will explicitly set this span's parent ID. If you are creating nested spans via `childOpts`, this will be handled automatically and you do not need to provide an ID.
+   * Defaults to the root span's ID.
+   * @param opts.childOpts An array containing options for direct children of the current span. Will create direct child spans for each set of options provided
+   */
+  constructor(opts: AddSpanOpts) {
+    const {startTimestamp, endTimestamp, op, description, status, problemSpan} = opts;
+
+    this.span = {
+      start_timestamp: startTimestamp,
+      timestamp: endTimestamp,
+      op,
+      description,
+      status: status ?? 'ok',
+      data: {},
+      // These values are automatically assigned by the TransactionEventBuilder when the spans are added
+      span_id: '',
+      trace_id: '',
+      parent_span_id: '',
+    };
+
+    this.problemSpan = problemSpan;
+  }
+
+  /**
+   *
+   * @param opts.numSpans If provided, will create the same span numSpan times
+   * @returns The MockSpan that was added. If numSpans > 1, will return the first new instance
+   */
+  addChild(opts: AddSpanOpts, numSpans = 1) {
+    const {startTimestamp, endTimestamp, op, description, status, problemSpan} = opts;
+
+    for (let i = 0; i < numSpans; i++) {
+      const span = new MockSpan({
+        startTimestamp,
+        endTimestamp,
+        op,
+        description,
+        status,
+        problemSpan,
+      });
+      this.children.push(span);
+    }
+
+    return this.children[0];
```

Comment:
Not that big of a deal, but I would have expected to just return `this` to allow method chaining like `span.addChild({...}).addChild({...}).addChild({...})`

Reply:
My thinking here was that this would make it easier to create deeply nested children, but maybe I could make this the default behaviour and implement a different function to create nested chains

Reply:
Changed it to return `this` instead 😄